servicetalk-http-api/src/main/java/io/servicetalk/http/api/DefaultHttpExecutionStrategy.java

```diff
@@ -82,7 +82,7 @@ public HttpExecutionStrategy merge(final HttpExecutionStrategy other) {
             return this;
         }
 
-        if (other instanceof SpecialHttpExecutionStrategy) {
+        if (!(other instanceof DefaultHttpExecutionStrategy)) {
```

Comment:
One of the big motivators for introducing this change was to unify the merge/difference process (remove special case instanceof and static equality checks). are any other changes required to make the simplification?